### PR TITLE
SASS-1543 Create 'how much do you want to claim for uniform, work clothes Amount page…

### PR DESCRIPTION
**Pull request by HarryGull**
Create 'how much do you want to claim for uniform, work clothes, or tools' amount EOY page
https://jira.tools.tax.service.gov.uk/browse/SASS-1543)

##### Before Reviewing
- [x]  Have you pulled the branch down?
- [x]  Have you assigned yourself to the PR?
- [x]  Have you moved the task to “in review” on JIRA?
- [x]  Have you checked to ensure all dependencies are up to date?
- [x]  Have you checked to ensure its been rebased against the current version of main?

##### Whilst Reviewing
- [x]  Have you run the tests?
- [x]  Have you run the journey tests?
- [x]  Have you looked at the JIRA story to make sure all Acceptance Criteria has been met?

##### After Reviewing
- [x]  Have you checked for merge conflicts or any changes in the current main that may affect the current pull request? i.e. does it need another rebase?
- [x]  Have you checked to make sure there are no builds in the pipeline before you merge?
- [ ]  Have you moved the task to “in pipeline” on Jira?

##### Before creating PR
- [x]  Have you run the tests?
- [x]  Have you run the journey tests? (where applicable)
- [x]  Have you addressed warnings where appropriate?
- [x]  Have you rebased against the current version of main?
- [x]  Have you checked code coverage isn’t lower than previously?

##### After PRs been raised
- [x]  Have you checked the PR Builder passes?

### CODE REVIEW

File: it/controllers/expenses/UniformsOrToolsExpensesAmountControllerISpec.scala [hunk 1]
```diff
@@ -0,0 +1,550 @@
...
+  val maxLimit: String = "100000000000"
+  val amountField = "#amount"
+  val amountFieldName = "amount"
+
+  private val userRequest = User(mtditid, None, nino, sessionId, affinityGroup)(fakeRequest)
+
+  private def expensesUserData(isPrior: Boolean, hasPriorExpenses: Boolean, expensesCyaModel: ExpensesCYAModel): ExpensesUserData =
+    ExpensesUserData(sessionId, mtditid, nino, taxYear - 1, isPriorSubmission = isPrior, hasPriorExpenses, expensesCyaModel)
+
+  def cyaModel(isUsingCustomerData: Boolean, expenses: Expenses): ExpensesCYAModel =
```
Comment (ina-tu) [inline it/controllers/expenses/UniformsOrToolsExpensesAmountControllerISpec.scala:46]: cyaModel can be removed since it's not being used

File: it/controllers/expenses/UniformsOrToolsExpensesAmountControllerISpec.scala [hunk 2]
```diff
@@ -0,0 +1,550 @@
...
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package controllers.expenses
+
+import controllers.employment.routes._
+import controllers.expenses.routes._
```
Comment (ina-tu) [inline it/controllers/expenses/UniformsOrToolsExpensesAmountControllerISpec.scala:20]: unused import

File: it/controllers/expenses/UniformsOrToolsExpensesAmountControllerISpec.scala [hunk 3]
```diff
@@ -0,0 +1,550 @@
...
+          userDataStub(userData(fullEmploymentsModel(hmrcExpenses = Some(employmentExpenses(fullExpenses)))), nino, taxYear)
+          insertExpensesCyaData(expensesUserData(isPrior = false, hasPriorExpenses = false,
+            fullExpensesCYAModel), userRequest)
+          authoriseAgentOrIndividual(user.isAgent)
+          urlGet(employmentExpensesAmountPageUrl(taxYear), user.isWelsh, follow = false, headers = Seq(HeaderNames.COOKIE -> playSessionCookies(taxYear)))
+        }
+
+        "has an SEE_OTHER(303) status" in {
+          result.status shouldBe SEE_OTHER
+          result.header("location") shouldBe Some(s"http://localhost:11111/income-through-software/return/$taxYear/view")
```
Comment (ina-tu) [inline it/controllers/expenses/UniformsOrToolsExpensesAmountControllerISpec.scala:278]: this can be changed to `Some(appConfig.incomeTaxSubmissionOverviewUrl(taxYear))`

File: it/controllers/expenses/UniformsOrToolsExpensesAmountControllerISpec.scala [hunk 4]
```diff
@@ -0,0 +1,550 @@
...
+          h1Check(user.specificExpectedResults.get.expectedHeading)
+          captionCheck(user.commonExpectedResults.expectedCaption)
+          textOnPageCheck(user.specificExpectedResults.get.expectedPreAmountParagraph(300), wantToClaimSelector)
+          buttonCheck(user.commonExpectedResults.continueButtonText, continueButtonSelector)
+          textOnPageCheck(user.commonExpectedResults.expectedCannotClaim, cannotClaimParagraphSelector)
+          textOnPageCheck(hintText, hintTextSelector)
+          textOnPageCheck(poundPrefixText, poundPrefixSelector)
+          inputFieldValueCheck("", amountField)
+          formPostLinkCheck(continueLink, formSelector)
+          welshToggleCheck(user.isWelsh)
```
Comment (ina-tu) [inline it/controllers/expenses/UniformsOrToolsExpensesAmountControllerISpec.scala:394]: missing tests for the no entry error messages

File: it/controllers/expenses/UniformsOrToolsExpensesAmountControllerISpec.scala [hunk 5]
```diff
@@ -0,0 +1,550 @@
...
+          h1Check(user.specificExpectedResults.get.expectedHeading)
+          captionCheck(user.commonExpectedResults.expectedCaption)
+          textOnPageCheck(user.specificExpectedResults.get.expectedPreAmountParagraph(300), wantToClaimSelector)
+          buttonCheck(user.commonExpectedResults.continueButtonText, continueButtonSelector)
+          textOnPageCheck(user.commonExpectedResults.expectedCannotClaim, cannotClaimParagraphSelector)
+          textOnPageCheck(hintText, hintTextSelector)
+          textOnPageCheck(poundPrefixText, poundPrefixSelector)
+          inputFieldValueCheck(maxLimit, amountField)
+          formPostLinkCheck(continueLink, formSelector)
+          welshToggleCheck(user.isWelsh)
```
Comment (ina-tu) [inline it/controllers/expenses/UniformsOrToolsExpensesAmountControllerISpec.scala:426]: missing tests for the over maximum error messages

File: it/controllers/expenses/UniformsOrToolsExpensesAmountControllerISpec.scala [hunk 6]
```diff
@@ -0,0 +1,550 @@
...
+          insertExpensesCyaData(expensesUserData(isPrior = false, hasPriorExpenses = false,
+            fullExpensesCYAModel), userRequest)
+          authoriseAgentOrIndividual(user.isAgent)
+          urlPost(employmentExpensesAmountPageUrl(taxYear), body = form, follow = false, welsh = user.isWelsh,
+            headers = Seq(HeaderNames.COOKIE -> playSessionCookies(taxYear)))
+        }
+
+        "has an SEE_OTHER(303) status" in {
+          result.status shouldBe SEE_OTHER
+          result.header("location") shouldBe Some(s"http://localhost:11111/income-through-software/return/$taxYear/view")
```
Comment (ina-tu) [inline it/controllers/expenses/UniformsOrToolsExpensesAmountControllerISpec.scala:524]: can be changed to Some(appConfig.incomeTaxSubmissionOverviewUrl(taxYear))

**Review by ina-tu — CHANGES_REQUESTED** [review]

**Review by ina-tu — APPROVED** [review]
